python-avd/pyavd/_eos_designs/structured_config/underlay/loopback_interfaces.py

```diff
@@ -21,84 +23,77 @@ class LoopbackInterfacesMixin(Protocol):
-
-            loopback0.update(isis_config)
-
-        loopback0 = {key: value for key, value in loopback0.items() if value is not None}
+                    loopback0.node_segment.ipv6_index = self._node_sid
 
-        loopback_interfaces.append(loopback0)
+        self.structured_config.loopback_interfaces.append(loopback0)
 
         # VTEP loopback
         if (
             self.shared_utils.overlay_vtep is True
             and self.shared_utils.vtep_loopback.lower() != "loopback0"
             and self.shared_utils.vtep_loopback.lower().startswith("lo")
         ):
-            vtep_loopback = {
-                "name": self.shared_utils.vtep_loopback,
-                "description": self.shared_utils.interface_descriptions.vtep_loopback_interface(
+            vtep_loopback = EosCliConfigGen.LoopbackInterfacesItem(
+                name=self.shared_utils.vtep_loopback,
+                description=self.shared_utils.interface_descriptions.vtep_loopback_interface(
                     InterfaceDescriptionData(
                         shared_utils=self.shared_utils, interface=self.shared_utils.vtep_loopback, description=self.inputs.vtep_loopback_description
                     )
                 ),
-                "shutdown": False,
-                "ip_address": f"{self.shared_utils.vtep_ip}/32",
-            }
+                shutdown=False,
+                ip_address=f"{self.shared_utils.vtep_ip}/32",
+            )
 
             if self.shared_utils.network_services_l3 is True and self.inputs.vtep_vvtep_ip is not None:
-                vtep_loopback["ip_address_secondaries"] = [self.inputs.vtep_vvtep_ip]
+                vtep_loopback.ip_address_secondaries.append_new(self.inputs.vtep_vvtep_ip)
 
             if self.shared_utils.underlay_ospf is True:
-                vtep_loopback["ospf_area"] = self.inputs.underlay_ospf_area
+                vtep_loopback.ospf_area = self.inputs.underlay_ospf_area
 
             if self.shared_utils.underlay_isis is True:
-                vtep_loopback.update({"isis_enable": self.shared_utils.isis_instance_name, "isis_passive": True})
+                vtep_loopback._update(isis_enable=self.shared_utils.isis_instance_name, isis_passive=True)
 
-            vtep_loopback = {key: value for key, value in vtep_loopback.items() if value is not None}
-
-            loopback_interfaces.append(vtep_loopback)
+            self.structured_config.loopback_interfaces.append(vtep_loopback)
 
         # Underlay Multicast RP Loopbacks
         if self.shared_utils.underlay_multicast_rp_interfaces is not None:
-            loopback_interfaces.extend(self.shared_utils.underlay_multicast_rp_interfaces)
-
-        return loopback_interfaces
+            for underlay_multicast_rp_interface in self.shared_utils.underlay_multicast_rp_interfaces:
+                loopback_item = EosCliConfigGen.LoopbackInterfacesItem(**underlay_multicast_rp_interface)
```

Comment:
Let's make sure the method in shared_utils return the proper object type instead.

Reply:
Done!